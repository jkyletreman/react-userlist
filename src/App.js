import React, { Component } from "react";

const userData = [
  {
    id: 1,
    first: "John",
    last: "Roberts",
    age: 51,
    location: "Chicago, IL",
    description: "John is a retired YouTuber"
  },
  {
    id: 2,
    first: "Frita",
    last: "Waters",
    age: 27,
    location: "Phoenix, AZ",
    description: "Frita loves the desert, cheese, and turtles."
  },
  {
    id: 3,
    first: "Bart",
    last: "Simpson",
    age: 24,
    location: "Springfield, IL",
    description: "Bart skateboards everywhere."
  }
];

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userSelected: ""
    };
  }

  handleClick = e => {
    this.setState({
      userSelected: e.target.innerText
    });
  };

  render() {
    const selectedUser = this.state.userIsSelected;

    const UserInfoFull = () =>
      <p></p>;

    return (
      <div>
        <ul>
          {userData.map(user => (
            <li key={user.id} onClick={this.handleClick}>
              {user.first} {user.last}
            </li>
          ))}
        </ul>
        <UserInfoFull />
      </div>
    );
  }
}

export default App;
