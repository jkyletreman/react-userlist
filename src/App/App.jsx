import React, { Component } from "react";
import UserList from "../UserList/UserList";
import UserInfoFull from "../UserInfoFull/UserInfoFull";
import UserData from "../context";

const App = () => {
  return (
    <UserData.Consumer>
      {context =>
        context.state.userIsSelected ? (
          <React.Fragment>
            <UserList
              userData={context.state.userData}
              selectUser={context.selectUser}
            />
            <UserInfoFull additionalInfo={context.state.additionalInfo} />
          </React.Fragment>
        ) : (
          <UserList
            userData={context.state.userData}
            selectUser={context.selectUser}
          />
        )
      }
    </UserData.Consumer>
  );
};

export default App
