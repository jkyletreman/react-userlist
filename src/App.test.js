import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe("<App />", () => {
  const wrapper = shallow(<App />)
  it('should render App', () => {
    // console.log(wrapper.debug())
  });
  it('should have a ul with 3 children', () => {
    expect(wrapper.find('ul').children().length).toBe(3);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<App />)
    expect(tree).toMatchSnapshot();
  });
})


// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });
