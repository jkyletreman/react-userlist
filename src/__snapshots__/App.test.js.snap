// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div>
  <ul>
    <li
      key="1"
      onClick={[Function]}
    >
      John
       
      Roberts
    </li>
    <li
      key="2"
      onClick={[Function]}
    >
      Frita
       
      Waters
    </li>
    <li
      key="3"
      onClick={[Function]}
    >
      Bart
       
      Simpson
    </li>
  </ul>
  <UserInfoFull
    selectedUser=""
  />
</div>
`;
